h2o-py/h2o/utils/shared_utils.py

```diff
@@ -353,6 +360,129 @@ def slice_is_normalized(s):
 check_lists_of_lists = _check_lists_of_lists
 
+gen_model_file_name = "h2o-genmodel.jar"
+h2o_predictor_class = "water.util.H2OPredictor"
+
+def find_file(name, path):
+    """
+    Helper function for predict_json() function to check if both MOJO and h2o-genmodel.jar are available
+    :param name: file name to find
+    :param path: folder/directory name with full path
+    :return: Successful path name or None if failed
+    """
+    for root, dirs, files in os.walk(path):
+        if name in files:
+            return os.path.join(root, name)
+
+def check_json(json_str):
+    try:
+        json_object = json.loads(json_str)
+    except ValueError:
+        raise RuntimeError("Error: Given JSON string does not look like valid JSON string.")
+    return True
+
+def predict_json(mojo_model, json, genmodelpath=None, labels=False, classpath=None, javaoptions='-Xmx4g', show_debug=False):
+    """
+    MOJO scoring function to take a pandas data frame as json string and use MOJO model as zip file to score
+
+    :param mojo_model: This is the MOJO model file name which is download after model was build using H2O, you can pass it two ways:
+           1. Full mojo model path i.e. /Users/avkashchauhan/src/github.com/h2oai/h2o-tutorials/tutorials/python_mojo_scoring/gbm_prostate_new.zip
+           2. mojo name first i.e. gbm_prostate_new.zip and genmodelpath = /Users/avkashchauhan/src/github.com/h2oai/h2o-tutorials/tutorials/python_mojo_scoring
+    :param json:  convert pandas frame to json (pd.to_json) and pass here
+    :param genmodelpath: This is the local file system folder name where both h2o-genmodel.jar and MOJO zip file is stored.
+    :param labels:     (Optional) True : Shows results, False: Does no show results and just pass result to given object
+    :param javaoptions: (Optional) These are the Java string options given by the user
+    :param show_debug: True/False - Default ( Use True to see the full Java command line)
+    :return: score as json values
+    """
+    # Verifying JSON
+    if show_debug:
+        print(json)
+
+    check_json(json)
+
+    # Checking java
+    java = H2OLocalServer._find_java()
+
+    jver_bytes = subprocess.check_output([java, "-version"], stderr=subprocess.STDOUT)
+    jver = jver_bytes.decode(encoding="utf-8", errors="ignore")
+    if show_debug:
+        print("  Java Version: " + jver.strip().replace("\n", "; "))
+    if "GNU libgcj" in jver:
+        raise H2OStartupError("Sorry, GNU Java is not supported for H2O.\n"
+                              "Please download the latest 64-bit Java SE JDK from Oracle.")
+    if "Client VM" in jver:
+        warn("  You have a 32-bit version of Java. H2O works best with 64-bit Java.\n"
+             "  Please download the latest 64-bit Java SE JDK from Oracle.\n")
+
+    # genmodelpath > must start and ends with "/" or add to it
+    gen_model_arg = ".:"
+
+    ## Working on mojo_model
+    mojo_model_path=None
+    head, tail = ntpath.split(mojo_model)
```

Comment:
Why ntpath (isn't ntpath for Windows paths?)? Should you use os.path instead?

Reply:
ntpath works good with both osx, linux and windows

Reply:
Please see the documentation of the ntpath module:

```
NAME
    ntpath - Common pathname manipulations, WindowsNT/95 version.

FILE
    /Users/mkurka/anaconda/lib/python2.7/ntpath.py

MODULE DOCS
    https://docs.python.org/library/ntpath

DESCRIPTION
    Instead of importing this module directly, import os and refer to this
    module as os.path.
```

It is better to use os.path (which is a platform-independent alias of the os-dependent implementation). None of our code uses ntpath.

Reply:
i have removed ntpath and bring back os.path